### DISCUSSION TITLE
How can to receive sms from android sms gateway to website?

### DISCUSSION THREAD

**Discussion by cilinjohnrey** [ibnux/Android-SMS-Gateway#23] [Q&A]
Hi ibnux, I want to fetch reply of the sent contact from android sms gateway to my website like inbox who contacts replied to my messages, How can i implement it?

**Comment by ibnux** [comment]
i never though about this, because i don't put contact on smartphone used for SMS Gateway.

i think need to emplement new command for push, maybe like,
```
to : phone number
text : contact
```

if app received **contact**, it will send to server the contact name. 

need to implement inside this function

https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/app/src/main/java/com/ibnux/smsgateway/layanan/PushService.java#L240-L274

how to implement
https://stackoverflow.com/questions/3712112/search-contact-by-phone-number

how to send to server
https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/app/src/main/java/com/ibnux/smsgateway/layanan/PushService.java#L84-L85

```java
SmsListener.sendPOST(getSharedPreferences("pref", 0).getString("urlPost", null), 
         arg1.getStringExtra("number"), contactName, "contact", context);
```

**Reply by cilinjohnrey** [reply]
Can i get the real status response of message if failed or sent or pending from android default sms app to android sms gateway then send to my website for message history?

**Comment by ibnux** [comment]
@longkoy732 you will get status on message body, as long you set URL for receiving SMS
type with **sent** and **delivered** have message content

https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/backend/index.php#L5-L7

type **delivered** has message content **success** or **failed**

type **sent** has message content **success** or **Generic failure** or **No service** or **Null PDU**  or **Radio off**

https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/app/src/main/java/com/ibnux/smsgateway/layanan/PushService.java#L70-L110
